src/modules/chat_freeze/index.js

```diff
@@ -1,12 +1,46 @@
 const $ = require('jquery');
+const watcher = require('../../watcher');
+const storage = require('../../storage');
 const twitch = require('../../utils/twitch');
 const keycodes = require('../../utils/keycodes');
 
 const CHAT_LIST_SELECTOR = '.chat-list';
 const MESSAGES_INDICATOR_SELECTOR = '.chat-list__more-messages';
 const FREEZE_KEYS = [keycodes.Ctrl, keycodes.Meta];
 
-let keysPressed = 0;
+const PATCHED_SYMBOL = Symbol();
+let twitchSetState;
+let bufferSize;
+
+function setChatBufferSize(size) {
+    if (!size) return;
+    try {
+        twitch.getChatController().chatBuffer.maxSize = size;
+    } catch (_) {}
+}
+
+function patchedSetState({isAutoScrolling}) {
+    if (isAutoScrolling !== undefined) {
+        setChatBufferSize(isAutoScrolling ? bufferSize : Infinity);
+    }
+    return twitchSetState.apply(this, arguments);
+}
+
+function patchScroller() {
```

Comment:
we should be able to do this without monkeypatching, so i'd rather do it without monkeypatching.

Reply:
If you don't monkey patch, and "click" on the `More messages below`, you're stuck with you're infite value, running into memory issues.

Reply:
I guess I could listen to the click on that element.

Reply:
No sometimes you can resume scrolling without clicking on the element. Monkeypatching seems to be the 'safe' way.
Should I drop this PR? @night

Reply:
i dont think you need to drop it.. just needs to be fixed to be safer.. you can detect when scrolling is happening with js events .. not sure why we need to patch to know if user is scrolling or not